src/compression.rs

```diff
@@ -196,6 +197,7 @@ impl Expr {
     }
     /// replaces the node at the end of the zipper with `new` prim,
     /// returning the new expression
+    #[allow(clippy::ptr_arg)]
```

Comment:
clippy wants zip: &[ZNode], because that's a more general type than Zip == Vec<ZNode>. However, I think this is less legible in this instance, so I've just disabled the warning.